stackslib/src/chainstate/stacks/db/unconfirmed.rs

```diff
@@ -852,18 +852,21 @@ mod test {
             // process microblock stream to generate unconfirmed state
             let sortdb = peer.sortdb.take().unwrap();
             peer.chainstate()
-                .reload_unconfirmed_state(&sortdb.index_conn(), canonical_tip.clone())
+                .reload_unconfirmed_state(&sortdb.index_handle_at_tip(), canonical_tip.clone())
                 .unwrap();
 
             let recv_balance = peer
                 .chainstate()
-                .with_read_only_unconfirmed_clarity_tx(&sortdb.index_conn(), |clarity_tx| {
-                    clarity_tx.with_clarity_db_readonly(|clarity_db| {
-                        clarity_db
-                            .get_account_stx_balance(&recv_addr.into())
-                            .unwrap()
-                    })
-                })
+                .with_read_only_unconfirmed_clarity_tx(
+                    &sortdb.index_handle_at_tip(),
```

Comment:
Same here -- let's use a fallible variant

Reply:
Does it matter here, since this is in a test?